client/extensions/woocommerce/app/promotions/promotion-models.js

```diff
@@ -108,46 +106,48 @@ const appliesToProductSale = {
 	labelText: translate( 'Applies to product' ),
 	cssClass: 'promotions__promotion-form-card-applies-to',
 	fields: {
-		appliesTo: {
-			component: (
-				<PromotionAppliesToField
-					selectionTypes={ [ { type: 'productIds' } ] }
-					singular={ true }
-				/>
-			),
-			// TODO: Remove this text after variable products are supported.
-			explanationText: translate(
-				'Note: Variable products cannot be selected.'
-			),
-		},
+		appliesTo: (
+			<PromotionAppliesToField
+				selectionTypes={ [ { type: 'productIds' } ] }
+				singular={ true }
+				// TODO: Remove this text after variable products are supported.
+				explanationText={ translate( 'Note: Variable products cannot be selected.' ) }
+			/>
+		),
 	},
 };
 
 /**
```

Comment:
The diffs below are a bit confusing, it might be better to look at the [new code by itself here](https://github.com/Automattic/wp-calypso/blob/c045a013f5b90a896e7ebf523ed6fb438fa88208/client/extensions/woocommerce/app/promotions/promotion-models.js#L120)